### DISCUSSION TITLE
Help resolving files with untranscribed gaps

### DISCUSSION THREAD

**Discussion by marielb** [deepgram/community#623] [General help] [answered]
### Which Deepgram product are you using?

Deepgram API

### Details

Hey Deepgram, I'm evaluating switching our platform over from AssemblyAI, but we are noticing gaps in the transcriptions. I'm wondering if there's anything that can be done to either fix the file or adjust the request to get these to work correctly. If not, are there some known limitations to be aware of, such as Deepgram having difficulty with audio that has background noise or music?

Here is one example (reproduced with https://playground.deepgram.com/):

Deepgram
> Overwatch 2 literally ruined the entire game in one singular update. If you were one of the 19 people still playing the game, you have even left, and this is all because they broke the hit boxes in the game. Take a look at how far your crosshair can be from somebody while still registering a shot. You can shoot towards **[missing transcription]** to play since his shurikens are literally going to land every time. Look how easy these right clicks land.
> 
> Hanzo isn't even a challenge, and random arrow shooting is almost Kiriko, Echo, shit. Even swordjorn has become unstoppable. They additionally gave every hero in the game a 50 HP boost, but that hardly even matters since it's so easy to kill people. Overwatch 2 was already in shambles ever since they announced PVE was canceled. And even after that, the game could just never find balance.
> 
> But this is just a shit storm of a game now that has no competent developers or any clue what direction the game should go in.

AssemblyAI
> Overwatch two literally ruined the entire game in one singular update. If you were one of the 19 people still playing the game, you have even left. And this is all because they broke the hitboxes in the game.
> 
> Take a look at how far your crosshair can be from somebody while still registering a shot. You can shoot towards **the next damn area code and this will still register as a full headshot. Now these hitbox changes apply to all projectile characters, so Genji has become super easy** to play since his shurikens are literally going to land every time.
> 
> Look how easy these right clicks land. Hanzo isn't even a challenge, and random arrow shooting is almost guaranteed to get hits. Always now Kiriko echo shit.
> 
> Even Sword Jordan has become unstoppable. They additionally gave every hero in the game a 50 hp boost, but that hardly even matters since it's so easy to kill people. Overwatch two is already in shambles ever since they announced PvE was canceled, and even after that the game could just never find balance.
> 
> But this is just a shitstorm of a game now that has no competent developers or any clue what direction the game should go in.

Deepgram request info (file `overwatch2cooked.mp4_(1)_V1.mp3.zip`)
```
{
  "metadata": {
    "transaction_key": "deprecated",
    "request_id": "c44f2a07-4fc8-436c-952f-af8626db79f5",
    "sha256": "f80bf0b5f08c4139089311385c4d7c385d4f26b70c6f44dabd693318f4a6ae5a",
    "created": "2024-02-26T22:47:42.068Z",
    "duration": 59.352,
    "channels": 1,
    "models": [
      "1abfe86b-e047-4eed-858a-35e5625b41ee"
    ],
    "model_info": {
      "1abfe86b-e047-4eed-858a-35e5625b41ee": {
        "name": "2-general-nova",
        "version": "2024-01-06.5664",
        "arch": "nova-2"
      }
    }
  },
...
}
```

Another example (file `transcript_with_missing_segments_V1.mp3.zip`):`request_id:"d4d53fbd-038b-4030-b7bc-59c0581a29b3"`

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

?smart_format=true&punctuate=true&paragraphs=true&language=en&model=nova-2

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

c44f2a07-4fc8-436c-952f-af8626db79f5

### If possible, please attach an example audio file to reproduce the issue.

[overwatch2cooked.mp4_(1)_V1.mp3.zip](https://github.com/deepgram/community/files/14412436/overwatch2cooked.mp4_.1._V1.mp3.zip)
[transcript_with_missing_segments_V1.mp3.zip](https://github.com/deepgram/community/files/14412441/transcript_with_missing_segments_V1.mp3.zip)

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jkroll-deepgram** [comment]
Hi @marielb, I see you're using our Nova-2 model. While that is our latest and generally most accurate model, we are improving some edge cases related to this behavior. To move forward with your testing right now, I recommend also looking at our Nova model, i.e. `model=nova` rather than `model=nova-2`.

**Reply by marielb** [reply]
Thanks for the response. `model=nova` does not seem to have the same issue on that audio file, but its WER isn't comparable to our existing providers. Do you have an idea of what types of input causes these issues? Would be great to have an understanding so that we can still enable Deepgram in some cases

**Reply by jkroll-deepgram** [reply]
@marielb  If you're open to looking at one more model version of ours, I'd also suggest trying a prior version of Nova-2 that shows a slightly different transcription pattern. For smart-formatted batch transcription, you can provide the params `model=nova-2&version=2023-11-14.3290&smart_format=true&paragraphs=true&language=en`. For reference, at present the latest version of Nova-2 is `version=2024-01-09.29447`. This still requires more investigation on our side to pinpoint the recent change, but I do see that this prior version doesn't miss the phrase in your example audio, and I've generally seen that this particular version performs well.

**Reply by marielb** [reply]
Hi @jkroll-deepgram. I tested out the `version=2023-11-14.3290` and it's still missing 0:32-35, which is fully audible in the file. For comparison:

Deepgram:
> Hanzo isn't even a challenge, and random arrow shooting is almost Kiriko, Echo, ...

AssemblyAI:
> Hanzo isn't even a challenge, and random arrow shooting is almost **guaranteed to get hits. Always now** Kiriko echo, ...

**Reply by jkroll-deepgram** [reply]
Hi @marielb, I appreciate your dedication to finding a Deepgram model and configuration that will work well for you. I've investigated further, and I am now thinking that these particular shortcomings are caused at least in part by an audio transcoding issue. 

You originally provided an mp3 file. I ran a simple terminal command to convert it to a wav file: `ffmpeg -i overwatch2cooked.mp3 -vn overwatch2cooked.wav`. I then found that our latest default `model=nova-2` transcribes the wav file with high accuracy and no missing segments. While Deepgram officially supports many different audio formats, this is a case where there seems to be an issue with mp3 that does not exist for wav. I hope this additional observation helps your use case.

**Reply by marielb** [reply]
Hi @jkroll-deepgram, thanks again for looking into this. I ran that ffmpeg command you sent over and did find that that fixed the transcription. Unfortunately the same fix didn't work for the second sample I attached, `transcript_with_missing_segments_V1.mp3.zip`. The best combination seems to be wav + version=2023-11-14.3290 but it's still missing some parts compared to the AssemblyAI transcription

Here are the results (beginning excerpt):

transcript_with_missing_segments_V1.mp3 (Deepgram, nova-2)
> My jokes. Yeah. As some may already have guessed, I'm Italian. And, of course, my research topic is about how we start we finished a lecture on Roman art, and we went around the city in Rome, and we saw a Roman sculpture like the, like the Augustus. We finished the lecture on the Roman architecture, and we went to see the Colosseum.

transcript_with_missing_segments_V1.wav (Deepgram, nova-2)
> Some of you may already know me. It's not because I'm famous. It's just because I asked a bunch of people to come and laugh at my jokes. Terrible. So far, it's been going great. Unfortunately, I want to thank that supporting my thesis, but if I have to eat another sandwich, I will die. For example, we start we finished a lecture on Roman art, and we went around the city in Rome, and we saw a Roman sculpture like the, at the Augustus. We finished the lecture on the Roman architecture, and we went to see the Colosseum.

transcript_with_missing_segments_V1.wav (Deepgram, nova-2&version=2023-11-14.3290)
> Hello, everyone. This is my first time as well, so I have a cheat sheet. Some of you may already know me. It's not because I'm famous. It's just because I asked a bunch of people to come and laugh at my jokes. Yeah. As some may already have guessed, I'm Italian. And, of course, my research topic is about how food here is terrible. So far, it's been going great. Unfortunately, I want to thank For example, we start we finished the lecture on Roman art, and we went around the city in Rome, and we saw, a Roman sculpture like the, at the Augustus. We finished the lecture on the Roman architecture, and we went to see the Colosseum.

transcript_with_missing_segments_V1.mp3 (AssemblyAI)
> This is my first time as well, so I have a cheat sheet, **not an exam. Thank you very much. So, thank you very much for coming. As I said, I'm Renato.** Some of you may already know me. It's not because I'm famous. It's just because I asked a bunch of people to come and laugh at my jokes. Yeah. As some may already have guessed, I'm italian. And of course, my research topic is about how food here is terrible. So far, it's been going great, **supporting my thesis. If I have to eat another sandwich, I will die. Actually, my background is in art history, and you may think that it's kind of weird studying art history in Italy.** For example, we finished a lecture on roman art, and we went around the city in Rome, and we saw a roman sculpture like the Augustus. We finished a lecture on roman architecture, and we went to see the Colosseum.

**Reply by jkroll-deepgram** [reply, marked as answer]
Hi @marielb, I just listened to the `transcript_with_missing_segments_V1` for the first time. That one does have some extenuating factors in that the speaker is far from the mic, and there is a lot of background laughter/voices/noise. However, you lay out a great case for how both the model and the audio encoding are contributing to the transcript quality. I'm coordinating across several teams internally so we can root-cause and take corrective action here. I'll keep following up with more information as I have it, but also feel free to ping me here as needed with more information or to get a new update.
